full_snapshot.proto

```diff
@@ -235,6 +236,67 @@ message TablespaceInformation {
   repeated string config = 4;   // Tablespace-level options, as "keyword=value" strings
 }
 
+message CollectorConfig {
+  bool enable_reports = 1;
+  bool disable_logs = 2;
+  bool disable_activity = 3;
+  bool enable_log_explain = 4;
+
+  string db_name = 5;
+  string db_username = 6;
+  string db_host = 7;
+  int32 db_port = 8;
+  string db_sslmode = 9;
+  string db_sslrootcert = 10;
+  string db_sslcert = 11;
+  string db_sslkey = 12;
```

Comment:
In the protobuf, it might be good to clarify that these are paths, i.e. not actual cert/key values.

Reply:
Actually, on second thought - this would be a good use case for the `has_` pattern - and either take the path or the contents. e.g. `has_db_sslkey = db_sslkey != "" || db_sslkey_contents != ""`

Reply:
Sure--maybe I'll do `db_has_...` to keep things consistent, but I like tracking contents as well.